core/domain/classifier_services.py

```diff
@@ -182,15 +209,33 @@ def handle_trainable_states(
         state_training_jobs_mapping.validate()
         state_training_jobs_mappings.append(state_training_jobs_mapping)
 
-    classifier_models.StateTrainingJobsMappingModel.create_multi(
-        state_training_jobs_mappings)
+    mapping_models = []
+    for state_training_job_mapping in state_training_jobs_mappings:
+        instance_id = (
+            classifier_models.StateTrainingJobsMappingModel.get_entity_id(
+                state_training_job_mapping.exp_id,
+                state_training_job_mapping.exp_version,
+                state_training_job_mapping.state_name
+            )
+        )
+        mapping_instance = classifier_models.StateTrainingJobsMappingModel(
+            id=instance_id, exp_id=state_training_job_mapping.exp_id,
+            exp_version=state_training_job_mapping.exp_version,
+            state_name=state_training_job_mapping.state_name,
+            algorithm_ids_to_job_ids=(
+                state_training_job_mapping.algorithm_ids_to_job_ids
+            ))
+
+        mapping_models.append(mapping_instance)
+    models_to_put.extend(mapping_models)
+    return models_to_put
 
 
 def handle_non_retrainable_states(
     exploration: exp_domain.Exploration,
     state_names: List[str],
     exp_versions_diff: exp_domain.ExplorationVersionsDiff
-) -> List[str]:
+) -> tuple(List[str], List[classifier_models.StateTrainingJobsMappingModel]):
```

Comment:
```suggestion
) -> Tuple[List[str], List[classifier_models.StateTrainingJobsMappingModel]]:
```

cc @sahiljoster32 I think that as we use `from __future__ import annotations` people can use built-in types (see https://mypy.readthedocs.io/en/stable/runtime_troubles.html#using-generic-builtins). Do we care, or do we want to forbid those?

Reply:
Hm, actually, we seem to be using "tuple" in several places from what I can tell. Lmk if I have to make this change here.

Reply:
@sahiljoster32 friendly ping!

Reply:
> cc @sahiljoster32 I think that as we use `from __future__ import annotations` people can use built-in types (see https://mypy.readthedocs.io/en/stable/runtime_troubles.html#using-generic-builtins). Do we care, or do we want to forbid those?

Ahh, till now the whole codebase is using typing's types ( like List, Tuple, and Set instead of a list, tuple, and set ). So, It would be great if we just stick to one notation rather than the mixed version of types.

Reply:
> Hm, actually, we seem to be using "tuple" in several places from what I can tell. Lmk if I have to make this change here.

Hi @kevintab95, can you please point me to those places where you have found `tuple` used for denoting the typing, I think we need to fix those places -- so that we have consistency all over the codebase.

Reply:
Thanks @sahiljoster32 for clarifying! I've updated it to use "Tuple" from typing package. The instances I saw in the codebase were in the return /arg types in comments.